thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/DataResource.java

```diff
@@ -227,33 +226,69 @@ public Long getMetricMaxDataTime(@PathParam("metricId") Long metricId) {
     return functions;
   }
 
+
   /**
-   * Returns list of AlertConfig object matching given name
-   * @param name
+   * Returns list of AnomalyFunction object matching given AlertConfigName
+   * @param alertName
    * @return
    */
   @GET
-  @Path("autocomplete/alert")
-  public List<AlertConfigDTO> getAlertsWhereNameLike(@QueryParam("name") String name) {
+  @Path("autocomplete/functionByAlertName")
+  public List<AnomalyFunctionDTO> getAlertsWhereAlertNameLike(@QueryParam("alertName") String alertName) {
     List<AlertConfigDTO> alerts = Collections.emptyList();
-    if (StringUtils.isNotBlank(name)) {
-      alerts = alertConfigDAO.findWhereNameLike("%" + name + "%");
+    if (StringUtils.isNotBlank(alertName)) {
+      alerts = alertConfigDAO.findWhereNameLike("%" + alertName + "%");
     }
-    return alerts;
+    return getFunctionsFromAlertConfigs(alerts);
   }
 
   /**
-   * Returns list of AlertConfig object matching given appName
+   * Returns list of AnomalyFunction object matching given appName
    * @param appname
    * @return
    */
   @GET
-  @Path("autocomplete/alertByAppname")
-  public List<AlertConfigDTO> getAlertsWhereAppNameLike(@QueryParam("appname") String appname) {
+  @Path("autocomplete/functionsByAppname")
```

Comment:
same here. use function or functions? lower case with - or camel style. Let's fix it. It will make it easier for other engineers to discover new endpoints.